DataFormats/CTPPSDigi/src/CTPPSPixelDigi.cc

```diff
@@ -0,0 +1,43 @@
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h"
+
+const uint32_t CTPPSPixelDigi::row_shift = 0; 
+const uint32_t CTPPSPixelDigi::column_shift = 11; 
+const uint32_t CTPPSPixelDigi::adc_shift = 22;
+const uint32_t CTPPSPixelDigi::row_width = 11; 
+const uint32_t CTPPSPixelDigi::column_width = 11; 
+const uint32_t CTPPSPixelDigi::adc_width = 10;
+const uint32_t CTPPSPixelDigi::row_mask = 0x7FF; 
+const uint32_t CTPPSPixelDigi::column_mask = 0x7FF;
+const uint32_t CTPPSPixelDigi::adc_mask = 0x3FF; 
+const uint32_t CTPPSPixelDigi::rowcol_mask = 0x3FFFFF;
+const uint32_t CTPPSPixelDigi::max_row = 0x7FF; 
+const uint32_t CTPPSPixelDigi::max_column = 0x7FF; 
+const uint32_t CTPPSPixelDigi::max_adc = 0x3FF;
+/// row_w 11, column_w 11, adc_w 10 bits
+
+void CTPPSPixelDigi::init(int row, int col, int adc) {
+
+  if(row_width+column_width+adc_width != 32) 
```

Comment:
Yes, sure - however there is no requirement to use all 32 bits when bit packing information. anyway, this could be replaced with a compile time check, eg

static_assert(row_width+column_width+adc_width > 32)

where you declare row_width, etc (since they are const statics)